Netable/Netable/Netable.swift

```diff
@@ -125,7 +125,15 @@ open class Netable {
                 switch decoded {
                 case .success(let raw):
                     let finalizedData = request.finalize(raw: raw)
-                    self.log(.requestSuccess(request: requestInfo, taskTime: time, statusCode: response.statusCode, responseData: data, finalizedResult: finalizedData))
+
+                    if case let .success(finalizedResult) = finalizedData {
+                        self.log(.requestSuccess(request: requestInfo, taskTime: time, statusCode: response.statusCode, responseData: data, finalizedResult: finalizedResult))
+                    } else if case let .failure(finalizedError) = finalizedData {
+                        throw NetableError.decodingError(finalizedError, data)
```

Comment:
Do we want to wrap this up in a decoding error? Decoding error is a wrapper mainly because it's holding a system error that happened during decoding, where if you pass the raw error through it would not be clear that the error happened in the decoding step. It seems like (if people are doing things right) this should already be a Netable resourceExtractionError (or at the very least, another known-to-the-user type, since any type here is thrown from user code), so it might not be as valuable to wrap it.

Reply:
Good point, updated